src/main/java/in/projecteka/consentmanager/user/SessionService.java

```diff
@@ -49,24 +50,19 @@
         if (StringUtils.isEmpty(request.getUsername()) || StringUtils.isEmpty(request.getPassword()))
             return Mono.error(ClientError.unAuthorizedRequest("Username or password is incorrect"));
 
-        var newLockedUser = new LockedUser(0, request.getUsername(), false, "", "");
-        Mono<LockedUser> lockedUser = lockedUserService.userFor(request.getUsername())
-                .switchIfEmpty(
-                        Mono.just(newLockedUser));
+        var newLockedUser = new LockedUser(0, request.getUsername(), false, "", new Date().toString());
 
-        return lockedUser
+        return lockedUserService.userFor(request.getUsername())
+                .switchIfEmpty(
+                        Mono.just(newLockedUser))
                 .flatMap(user -> {
                     if (lockedUserService.isUserBlocked(user)) {
-                        return Mono.error(ClientError.userBlocked());
+                        return lockedUserService.validateAndUpdate(user).then(Mono.error(ClientError.userBlocked()));
```

Comment:
If the user is Blocked, why are we validating again? Also what are we updating?

Reply:
we are updating the invalid_attempts